05-JWT-Basics/preferred/controllers/index.js

```diff
@@ -0,0 +1,22 @@
+const jwt = require("jsonwebtoken");
+
+const logon = async (req, res) => {
+  const { userName, password } = req.body;
+
+  const secret = process.env.SECRET;
+  const lifeTime = process.env.LIFETIME;
+
+  const token = jwt.sign({ userName }, secret, {
+    expiresIn: lifeTime,
+  });
+
+  return res.status(200).json({ token });
+};
```

Comment:
In this method, you'll want some logic to ensure that the request body included the expected fields. In other words, throw a `400 Bad Request` error if a user makes a request without the `userName` or `password` field. (I used `username` the first time, and that lead to the `welcome` method printing `Welcome, undefined`.)